python/ray/tests/test_asyncio_cluster.py

```diff
@@ -0,0 +1,34 @@
+# coding: utf-8
+import asyncio
+import sys
+
+import pytest
+import numpy as np
+
+import ray
+from ray.cluster_utils import Cluster
+
+
+@pytest.mark.asyncio
+async def test_asyncio_cluster_wait():
+    cluster = Cluster()
+    head_node = cluster.add_node()
+    cluster.add_node(resources={"OTHER_NODE": 100})
+
+    ray.init(address=head_node.address)
+
+    @ray.remote(num_cpus=0, resources={"OTHER_NODE": 1})
+    def get_array():
+        return np.random.random((192, 1080, 3)).astype(np.uint8)  # ~ 0.5MB
+
+    object_ref = get_array.remote()
+
+    await asyncio.wait_for(object_ref, timeout=10)
+
+    ray.shutdown()
+    cluster.shutdown()
+
```

Comment:
Is this guaranteed to fail without this fix? Or do you need to run it more times?

Reply:
It's guaranteed to fail without `SubscribeWaitDependencies`. It's a little more flakey with the Get() we had originally but we guarded them with `Contains` now.